zboxcore/sdk/writemarker_mutex.go

```diff
@@ -60,257 +52,215 @@ func CreateWriteMarkerMutex(client *client.Client, allocationObj *Allocation) (*
 
 	return &WriteMarkerMutex{
 		allocationObj: allocationObj,
-		zbox:          sdks.New(client.ClientID, client.ClientKey, client.SignatureScheme, client.Wallet),
 	}, nil
 }
 
-// Lock acquire WriteMarker lock from blobbers
-func (m *WriteMarkerMutex) Lock(ctx context.Context, connectionID string) error {
-	if m == nil {
-		return errors.Throw(constants.ErrInvalidParameter, "WriteMarkerMutex")
-	}
-	m.mutex.Lock()
-	defer m.mutex.Unlock()
-
-	if m.allocationObj == nil {
-		return errors.Throw(constants.ErrInvalidParameter, "allocationObj")
-	}
+func (wmMu *WriteMarkerMutex) Unlock(
+	ctx context.Context, mask zboxutil.Uint128,
+	blobbers []*blockchain.StorageNode,
+	timeOut time.Duration, connID string,
```

Comment:
The timeout could be attached to the ctx, so user could just wrap it as `ctx, cancel := context.WithTimeout(ctx, timeout)`